external/error_report/error_report.c

```diff
@@ -0,0 +1,205 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+#include <stdio.h>
+#include <stdlib.h>
+#include <tinyara/config.h>
+#include <time.h>
+#include <pthread.h>
+#include <unistd.h>
+#include <error_report/error_report.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <sched.h>
+#include "sched/sched.h"
+#include <libgen.h>
+
+#define ERR_IP_ADRR_LEN 16
+
+#ifdef CONFIG_ERROR_REPORT_DEBUG
+#define rt_errdebug printf
+#else
+#define rt_errdebug(...)
+#endif
+
+_rt_error_data_s g_error_report[CONFIG_ERROR_REPORT_NENTRIES];
+typedef enum {
+	ERRSTATE_UNINITIALIZED = -1,
+	ERRSTATE_INITIALIZED,
+	ERRSTATE_MEMUNDERFLOW,
+	ERRSTATE_BUFFER_EXCEEDED,
+} err_fsm_e;
```

Comment:
We usaually use ```_t``` for typedef. ```_e``` for enum without typedef.
_s same.

Reply:
Changed according to standards. Thank you.